src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -454,7 +451,7 @@ private static void StelemRef(object?[] array, nint index, object? obj)
                 goto notExactMatch;
 
             doWrite:
-                WriteBarrier(ref element, obj);
+                RuntimeHelpers.WriteBarrierUnchecked(ref element, obj);
```

Comment:
The naming used for write barriers is "WriteBarrier" and "CheckedWriteBarrier". I do not think introducing "WriteBarrierUnchecked" as an alternative name for a regular "WriteBarrier" is helping clarity.

Reply:
Ok, renamed back to WriteBarrier